### DISCUSSION TITLE
Attribute Error while training a basic MACE model.

### DISCUSSION THREAD

**Discussion by RISHAV01CODER** [ACEsuit/mace#552] [Q&A] [answered]
I was trying to implement the example code in the tutorial but am facing a problem - 

My script:  C:\> python .\Macerepo\mace\mace\cli\run_train.py --name="Mace_model" --train_file="C:\botnetdata\BOTNet-datasets\dataset_3BPA\tra
in_300K.xyz" --valid_fraction=0.05 --test_file="C:\botnetdata\BOTNet-datasets\dataset_3BPA\test_300K.xyz" --E0s='{1:-13.6631812922312
26, 6:-1029.2809654211628, 7:-1484.1187695035828, 8:-2042.0330099956639}' --model="ScaleShiftMACE" --hidden_irreps='32x0e' --r_max=4.
0 --batch_size=2 --max_num_epochs=4 --ema --ema_decay=0.99 --amsgrad --default_dtype="float32" --device=cpu --seed=123 --swa 
(all commands written in a single line).

Error showing: AttributeError: 'Namespace' object has no attribute 'beta'.

I am using windows powershell.
Link to the full output which was displayed on the powershell - [https://docs.google.com/document/d/1Op_i95U6347nG9udBzBEDdejJXCNnrAHSx1yma_27eI/edit](url)

and below is a snippet of the code block in run_train.py file:

![image](https://github.com/user-attachments/assets/0f13f33a-8f86-4753-864f-6c805219d4ba)

Can anyone pls suggest what is wrong here? What should I do?

**Comment by ilyes319** [comment]
hey @RISHAV01CODER ,

What pytorch version are you using? Also what is your MACE version?

**Reply by ilyes319** [reply, marked as answer]
I am seeing your are using mace 0.3.5 can you please make sure to update to the latest version which is 0.3.6. Please do `pip install --upgrade mace-torch`

**Reply by RISHAV01CODER** [reply]
Yes, it's working. The model is training now it seems. Thanks for the help @ilyes319 !

However, I was getting a runtime error. Can it be due to the limitation of memory in my computer? I have attached the output here.

(RuntimeError: File Mace_model.model cannot be opened.)
[mace run 28aug24.pdf](https://github.com/user-attachments/files/16797228/mace.run.28aug24.pdf)

**Reply by ilyes319** [reply]
Are you on a windows machine?

**Reply by RISHAV01CODER** [reply]
Yes, I'm using windows. Mine is a HP i7- 16GB RAM system.

**Reply by ilyes319** [reply]
Then it is a path problem. Try naming your model something different. But this is not really a problem, you already have the model saved and training complete.

**Reply by RISHAV01CODER** [reply]
Ok sir. Thank you very much!

**Comment by RISHAV01CODER** [comment]
Hello Sir,
I'm using the torch version - 2.3.1+cu118. And MACE version 0.3.5 .

Regards,
Rishav Mitra

**Comment by RISHAV01CODER** [comment]
Ok sir, I'll do it. Thanks!

Regards,
Rishav M.
